java/vector/src/main/java/org/apache/arrow/vector/DateMilliVector.java

```diff
@@ -127,9 +129,7 @@ public LocalDateTime getObject(int index) {
       return null;
     } else {
       final long millis = valueBuffer.getLong(index * TYPE_WIDTH);
-      final LocalDateTime localDateTime = new org.joda.time.LocalDateTime(millis,
-              org.joda.time.DateTimeZone.UTC);
-      return localDateTime;
+      return DateUtility.getLocalDateTimeFromEpochMilli(millis);
```

Comment:
This change here is a change in API. It will break all applications using this getObject API. In the first impression it didn't look like API change because it is returning LocalDateTime, but this LocalDateTime is java.time and not org.joda.time.